static/js/components/CandidatesPreferences.jsx

```diff
@@ -0,0 +1,645 @@
+import * as profileActions from "../ducks/profile";
+import Responsive from "./Responsive";
+import FoldBox from "./FoldBox";
+import FormValidationError from "./FormValidationError";
+import { allowedClasses } from "./ClassificationForm";
+import ScanningProfilesList from "./ScanningProfilesList";
+
+dayjs.extend(utc);
+
+const rejectedStatusSelectOptions = [
+  { value: "hide", label: "Hide rejected candidates" },
+  { value: "show", label: "Show rejected candidates" },
+];
+
+const savedStatusSelectOptions = [
+  { value: "all", label: "regardless of saved status" },
+  { value: "savedToAllSelected", label: "and is saved to all selected groups" },
+  {
+    value: "savedToAnySelected",
+    label: "and is saved to at least one of the selected groups",
+  },
+  {
+    value: "savedToAnyAccessible",
+    label: "and is saved to at least one group I have access to",
+  },
+  {
+    value: "notSavedToAnyAccessible",
+    label: "and is not saved to any of group I have access to",
+  },
+  {
+    value: "notSavedToAnySelected",
+    label: "and is not saved to any of the selected groups",
+  },
+  {
+    value: "notSavedToAllSelected",
+    label: "and is not saved to all of the selected groups",
+  },
+];
+
+const useStyles = makeStyles((theme) => ({
+  filterListContainer: {
+    padding: "1rem",
+    display: "flex",
+    flexFlow: "column nowrap",
+  },
+  formRow: {
+    margin: "1rem 0",
+    "& > div": {
+      width: "100%",
+    },
+  },
+  redshiftField: {
+    display: "inline-block",
+    marginRight: "0.5rem",
+  },
+  savedStatusSelect: {
+    margin: "1rem 0",
+    "& input": {
+      fontSize: "1rem",
+    },
+  },
+  annotationSorting: {
+    "& label": {
+      marginTop: "1rem",
+    },
+    "& div": {
+      width: "100%",
+    },
+  },
+  saveButton: {
+    marginTop: "1rem",
+  },
+  dialogContent: {
+    backgroundColor: theme.palette.background.default,
+  },
+  header: {
+    justifyContent: "space-between",
+  },
+}));
+
+function getStyles(classification, selectedClassifications, theme) {
+  return {
+    fontWeight:
+      selectedClassifications.indexOf(classification) === -1
+        ? theme.typography.fontWeightRegular
+        : theme.typography.fontWeightMedium,
+  };
+}
+
+// eslint-disable-next-line react/display-name
+const Transition = React.forwardRef((props, ref) => (
+  // eslint-disable-next-line react/jsx-props-no-spreading
+  <Slide direction="up" ref={ref} {...props} />
+));
+
+const CandidatesPreferences = ({
+  selectedScanningProfile,
+  setSelectedScanningProfile,
+}) => {
+  const preferences = useSelector((state) => state.profile.preferences);
+  const availableAnnotationsInfo = useSelector(
+    (state) => state.candidates.annotationsInfo
+  );
+  const classes = useStyles();
+  const theme = useTheme();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    // Grab the available annotation fields for filtering
+    if (!availableAnnotationsInfo) {
```

Comment:
Doesn't this also get fetched on the main candidate list component? Could this then result in two fetches rather than just one?